pycaret/time_series/forecasting/oop.py

```diff
@@ -1063,6 +1063,39 @@ def _check_setup_ran(self):
             raise RuntimeError("This function requires the users to run setup() first.\
                 More info: https://pycaret.gitbook.io/docs/get-started/quickstart")
 
+    def _mlflow_log_setup(self, plots=['diagnostics', 'decomp', 'diff']):
+        self.logger.info("Creating MLFlow logs for plots that does not requre an estimator at the setup stage")
+
+        # import mlflow
+        import mlflow
+        import mlflow.sklearn
+
+        mlflow.set_experiment(self.exp_name_log)
+
+        with mlflow.start_run(nested=True) as run:
+
+            self.logger.info(
+                    "Begin logging diagnostics, decomp, and diff plots ================"
+                )
+
+            def _log_plot(plot):
+                try:
+                    plot_name = self.plot_model(
+                        plot=plot,
+                        return_fig=True,
+                    )
+                    mlflow.log_artifact(plot_name)
+                    os.remove(plot_name)
```

Comment:
In a future PR, we should look to abstract this out into its own function so that it can be reused between `_mlflow_log_model` and `_mlflow_log_setup`